dashboard/app/mailers/teacher_mailer.rb

```diff
@@ -2,9 +2,18 @@ class TeacherMailer < ActionMailer::Base
   default from: 'Hadi Partovi <hadi_partovi@code.org>'
 
   # Send newly registered teachers a welcome email
-  def new_teacher_email(teacher)
+  def new_teacher_email(teacher, teacher_locale = 'en-US')
     @teacher = teacher
-    mail to: teacher.email, subject: I18n.t('teacher_mailer.new_teacher_subject')
+
+    # We currently have two flavors for the new teacher welcome email,
+    # en-US (the default) and es-MX.
+    email_locale = teacher_locale.to_s == 'es-MX' ? teacher_locale : 'en-US'
+    email_template = teacher_locale.to_s == 'es-MX' ? 'new_teacher_email_es-MX' : 'new_teacher_email'
```

Comment:
With this logic, how easy is it to expand to more languages? For example, es-es is actually more used than es-mx, so I could see us adding a template for that pretty soon. And as we actively pursue partnerships, I imagine we will want to add more languages' templates in the near future.

Reply:
It's quite easy. We can use a simple mapping like `user_locale -> (email_locale, teacher_locale)` when we have more cases.